core/include/gnuradio-4.0/Settings.hpp

```diff
@@ -518,6 +529,29 @@ class CtxSettings : public SettingsBase {
         }
     }
 
+    [[nodiscard]] NO_INLINE const SettingsCtx& activeContext() const noexcept override { return _activeCtx; }
+
+    [[nodiscard]] NO_INLINE bool removeContext(SettingsCtx ctx) override {
+        if (ctx.time == 0ULL) {
+            ctx.time = settings::convertTimePointToUint64Ns(std::chrono::system_clock::now());
+#ifdef __EMSCRIPTEN__
+            ctx.time += _timePrecisionTolerance;
+#endif
+        }
+
+        std::optional<SettingsCtx> bestMatchSettingsCtx = findBestMatchSettingsCtx(ctx);
+        if (!bestMatchSettingsCtx) {
+            return false;
+        }
+
+        if (_activeCtx.context == ctx.context) {
+            activateContext(); // Activate default context
+        }
+        _storedParameters.erase(ctx.context);
+
+        return true;
+    }
```

Comment:
We should compare the context for removal with an exact match by name and time. 
Additionally, we should not allow the default context to be removed.

Example of exact match

```cpp
bool findExactMatch(const SettingsCtx& ctx) const {
    auto it = _storedParameters.find(ctx.context);
    if (it != _storedParameters.end()) {
        const auto& vec = it->second;
        return std::ranges::any_of(vec, [&ctx](const auto& pair) {
            return pair.first.time == ctx.time;
        });
    }
    return false;
}
```